### ISSUE TITLE
Running pretty_errors as script with virtualenv does not find install folders

### ISSUE THREAD

**Issue by KittyKittyKitKat** [onelivesleft/PrettyErrors#39]
I'm on Ubuntu 20.04 and attempting to install pretty_errors using the `python -m pretty_errors -s -p` command within a virtualenv. 
Upon running the command, I get the following output:
```
To have pretty_errors be used when you run any python file you may add it to your usercustomize.py (user level) 
or sitecustomize.py (system level), or to pretty_errors.pth.

(just hit <enter> to accept the defaults if you are unsure)


 Choose folder to install into:

0: Exit

Option: [default: 0]
```
After which I hit enter and the install fails.
Investigating the `__main__.py` file along with Issue #38 lead me to the following function on line 41
```py
def getsitepackages():
    pattern1 = re.compile(r'^%s$' % re.escape(os.path.join(os.environ['VIRTUAL_ENV'], 'lib', 'python[0-9.]+', 'site-packages')))
    pattern2 = re.compile(r'^%s$' % re.escape(os.path.join(os.environ['VIRTUAL_ENV'], 'lib', 'site-packages')))
    paths = [path for path in set(sys.path) if pattern1.search(path) or pattern2.search(path)]
    return paths
```
Using some basic debugging, I discovered that the `paths` variable returned was completely empty, despite the fact that `pattern1` should have match my virtualenv path: `/home/username/.virtualenvs/general/lib/python3.9/site-packages`
I traced the source of this error to the call to `re.escape` in `pattern1`. The `os.path.join` call returns `/home/username/.virtualenvs/general/lib/python[0-9.]+/site-packages` as expected, but the call to `re.escape` returns the string `'/home/username/\\.virtualenvs/general/lib/python\\[0\\-9\\.\\]\\+/site\\-packages'`. Because `re.escape` escapes *all* regex special characters, the `python[0-9]+` portion of the regex gets mangled and causes `pattern1.search` to fail. Omitting the call to `re.escape` matches the string as expected. Upon removing the call to `re.escape` and running `python -m pretty_errors -s -p` again, everything installed and functioned as expected.

```py
>>> import os, sys, re
>>> sys.path
['', '/usr/lib/python39.zip', '/usr/lib/python3.9', '/usr/lib/python3.9/lib-dynload', '/home/username/.virtualenvs/general/lib/python3.9/site-packages']
>>> venv_path = sys.path[-1]
>>> os.path.join(os.environ['VIRTUAL_ENV'], 'lib', 'python[0-9.]+', 'site-packages')
'/home/username/.virtualenvs/general/lib/python[0-9.]+/site-packages'
>>> pattern1_escape = re.compile(r'^%s$' % re.escape(os.path.join(os.environ['VIRTUAL_ENV'], 'lib', 'python[0-9.]+', 'site-packages')))
>>> pattern1_escape.search(venv_path)
>>> pattern1_no_escape = re.compile(r'^%s$' % os.path.join(os.environ['VIRTUAL_ENV'], 'lib', 'python[0-9.]+', 'site-packages'))
>>> pattern1_no_escape.search(venv_path)
<re.Match object; span=(0, 61), match='/home/username/.virtualenvs/general/lib/python3.9/s>
>>> pattern1_no_escape.search(venv_path).group(0)
'/home/username/.virtualenvs/general/lib/python3.9/site-packages'
```

Here is my `pyvenv.cfg` file for Python and virtualenv info:
```cfg
home = /usr
implementation = CPython
version_info = 3.9.5.final.0
virtualenv = 20.4.6
include-system-site-packages = false
base-prefix = /usr
base-exec-prefix = /usr
base-executable = /usr/bin/python3.9
```

**Comment by onelivesleft** [comment]
Thanks for the detailed breakdown of the problem, I'll put it on my todo

**Comment by gerardpc** [comment]
I was coming to report the exact same issue as @KittyKittyKitKat

**Comment by onelivesleft** [comment]
This should be fixed in v1.2.22
